commands/permissions.go

```diff
@@ -0,0 +1,47 @@
+package commands
+
+import (
+	"context"
+
+	"github.com/UCCNetworkingSociety/Netsoc-Discord-Bot/config"
+	"github.com/UCCNetworkingSociety/Netsoc-Discord-Bot/logging"
+	"github.com/bwmarrin/discordgo"
+)
+
+// IsAllowed determines if the given user has a role set out in the permissions config
+func IsAllowed(ctx context.Context, s *discordgo.Session, authorID string, permissionKey string) bool {
+	l, loggerOk := logging.FromContext(ctx)
+	conf := config.GetConfig()
+	member, _ := s.GuildMember(conf.GuildID, authorID)
```

Comment:
Probably shouldn't ignore the error here :)
https://godoc.org/github.com/bwmarrin/discordgo#Session.GuildMember